### PR TITLE
Release v0.18.0

### PR DESCRIPTION
**Pull request by dhrgit**
## Reason for This PR

Release v0.18.0.

## Description of Changes

- Fixed some typos left behind in `docs/vsock.md`.
- Updated `CHANGELOG.md` with all user-facing changes since v0.17.0.
- Fixed `CHANGELOG.md` (line length) formatting.
- Updated Firecracker version via `devtool prepare_release`.

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

- [x] All commits in this PR are signed (`git commit -s`).
- [x] Either this PR is linked to an issue, or, the reason for this PR is
      clearly provided. 
- [x] The description of changes is clear and encompassing.
- [x] Either no docs need to be updated as part of this PR, or, the required
      doc changes are included in this PR. Docs in scope are all `*.md` files
      located either in the repository root, or in the `docs/` directory.
- [x] Either no code has been touched, or, code-level documentation for touched
      code is included in this PR.
- [x] Either no API changes are included in this PR, or, the API changes are
      reflected in `firecracker/swagger.yaml`.
- [x] Either the changes in this PR have no user impact, or, the changes in
      this PR have user impact and have been added to the `CHANGELOG.md` file.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,9 +9,12 @@
 
 ### Fixed
```
Comment (serban300) [inline CHANGELOG.md]: Also fixed seccomp edge case: https://github.com/firecracker-microvm/firecracker/issues/1206 if it's worth mentioning.

Comment (dhrgit) [reply CHANGELOG.md]: Added.

File: CREDITS.md [hunk 2]
```diff
@@ -72,16 +80,21 @@ Contributors to the Firecracker repository:
...
 * Robert Grimes <rmzgrimes@gmail.com>
+* Rodrigue Chakode <rodrigue.chakode@continental-corporation.com>
 * Rolf Neugebauer <neugebar@amazon.com>
 * Sam Jackson <sam@clique.app>
 * Sean Lavine <freewil@users.noreply.github.com>
 * Serban Iorga <seriorga@amazon.com>
+* Shen Jiale <shenjiale@baidu.com>
 * Sripracha <ramsri@amazon.com>
+* Tamio-Vesa Nakajima <tamiove@amazon.com>
+* Tamio-Vesa Nakajima <tnk@amazon.com>
```
Comment (aghecenco) [inline CREDITS.md]: This is a duplicate entry. Please update `.mailmap` to merge the 2 aliases for Tamio and remove the duplicate from the contributors list.

File: CHANGELOG.md [hunk 3]
```diff
@@ -9,9 +9,14 @@
...
 
-- Corrected firecracker-experimental.yaml indentation issues that
-  prevented code generation.
 - Updated the documentation for integration tests.
+- Fixed high CPU usage before guest network interface is brought up (GitHub
```
[the remark this answers is not in the document]
Comment (acatangiu) [reply CHANGELOG.md]: Hmm.. I think it'd be cool to have the issue number here so people can easily find it for more details.

But like @dianpopa said, if we do this here we need to be consistent and add it for every `Fixed` entry and every `Added` entry where we can reference an issue.

File: CREDITS.md (hunk 2 above)
Comment (dhrgit) [reply CREDITS.md]: Fixed.

File: CHANGELOG.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (dhrgit) [reply CHANGELOG.md]: I found mentioning the GitHub issue helpful for traceability reasons. The value that I see added is that one may look up more detailed information pertaining to a particular change (i.e. fix/change/addition). Without the GitHub reference, I find gathering that information to be much more challenging.

[the remark this answers is not in the document]
Comment (acatangiu) [reply CHANGELOG.md]: I'm fine with having it, but let's be consistent. At the very least, the
> - New device: virtio-vsock, backed by Unix domain sockets. See `docs/vsock.md`.

entry needs the GitHub issue as well. A lot of the early design details are in issue #650

**Comment by acatangiu** [comment]
Also rebase on top of master since you're changing files anyway (save some CI time).

File: CHANGELOG.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (dhrgit) [reply CHANGELOG.md]: Added.

**Pull request merged**